images/macos/helpers/Xcode.Installer.psm1

```diff
@@ -0,0 +1,212 @@
+Import-Module "$PSScriptRoot/Xcode.Helpers.psm1"
+
+function Install-XcodeVersion {
+    param(
+        [Parameter(Mandatory)]
+        [string]$Version,
+        [Parameter(Mandatory)]
+        [string]$LinkTo
+    )
+
+    $xcodeDownloadDirectory = "$env:HOME/Library/Caches/XcodeInstall"
+    $xcodeTargetPath = Get-XcodeRootPath -Version $LinkTo
+
+    Invoke-DownloadXcodeArchive -Version $Version
+    Expand-XcodeXipArchive -DownloadDirectory $xcodeDownloadDirectory -TargetPath $xcodeTargetPath
+
+    Get-ChildItem $xcodeDownloadDirectory | Remove-Item -Force
+}
+
+function Invoke-DownloadXcodeArchive {
+    param(
+        [Parameter(Mandatory)]
+        [string]$Version
+    )
+
+    $resolvedVersion = Resolve-ExactXcodeVersion -Version $Version
+    if (-not $resolvedVersion) {
+        throw "Version '$Version' can't be matched to any available version"
+    }
+
+    # TO-DO: Consider replacing of xcversion with own implementation
+    Write-Host "Downloading Xcode $resolvedVersion"
+    Invoke-ValidateCommand "xcversion install '$resolvedVersion' --no-install"
+}
+
+function Resolve-ExactXcodeVersion {
+    param(
+        [Parameter(Mandatory)]
+        [string]$Version
+    )
+
+    # if toolset string contains spaces, consider it as a full name of Xcode
+    if ($Version -match "\s") {
+        return $Version
+    }
+
+    $semverVersion = [SemVer]::Parse($Version)
+    $availableVersions = Get-AvailableXcodeVersions
+    $satisfiedVersions = $availableVersions | Where-Object { $semverVersion -eq $_.stableSemver }
+    return $satisfiedVersions | Select-Object -Last 1 -ExpandProperty rawVersion
+}
+
+function Get-AvailableXcodeVersions {
+    $rawVersionsList = & xcversion list | ForEach-Object { $_.Trim() } | Where-Object { $_ -match "^\d" }
```

Comment:
Why didn't you add the `xcversion update` command? As far as I know, the output of `xcversion list` can be outdated without running this command.

Reply:
It can't be outdated because `xcversion` was just installed and don't have local cache so `xcversion list` will update list.